src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -125,36 +126,73 @@ public void cancelOperation_usingAccelerator() {
 
         //Edit Alice to change to new values
         EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
-        editPersonDialog.enterNewValues(newAlice);
-        editPersonDialog.pressEnter();
-
+        assertTrue(editPersonDialog.isShowingPerson(td.alice));
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
         personListPanel.clickOnPerson(newAlice);
         assertEquals(alicePersonCard, newAlice);
         personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         assertEquals(alicePersonCard, td.alice);
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
         assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] was cancelled.");
 
         //New
         EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
         Person pandaWong = new PersonBuilder("Panda", "Wong")
                 .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
                 .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
-        addPersonDialog.enterNewValues(pandaWong);
-        addPersonDialog.clickOk();
+        addPersonDialog.enterNewValues(pandaWong).clickOk();
```

Comment:
Same here. I prefer to have Edit, Delete, and New in three separate tests. When we try to test too many things at once, some subtle test cases can get left out.